ut2004

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# Copyright 2023 Daniel Rusek <mail@asciiwolf.com>
+# Based on ioquake3 launcher script by Mattias Bengtsson
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+function error-message {
+    cat << EOM
+Unreal Tournament 2004 needs the official game files
+to function properly.
+
+Please copy the game installation directory content
+from Steam, GOG or retail DVD release here:
+
+    ~/.var/app/com.epicgames.ut2004/data/
+
+… and restart Unreal Tournament 2004
+EOM
+}
+
+for i in Animations Sounds System Textures; do
+    if [ ! -d "${XDG_DATA_HOME}/${i}" ]; then
+        zenity --error      \
+               --width=400  \
+               --height=200 \
+               --text="$(error-message)"
+        exit 2
+    fi
+done
+
+if [ ! -f "${XDG_DATA_HOME}/patched" ]; then
+    cp -rf /app/extra/UT2004-Patch/* "${XDG_DATA_HOME}"/
+    # Replace outdated bundled libraries with newer versions.
+    cp -L /usr/lib/x86_64-linux-gnu/libopenal.so.1 "${XDG_DATA_HOME}"/System/openal.so
+    cp -L /app/lib/libSDL-1.2.so.0 "${XDG_DATA_HOME}"/System/libSDL-1.2.so.0
+#    cp -L /app/lib/libstdc++.so.5 "${XDG_DATA_HOME}"/System/libstdc++.so.5
+    sed -i "s/ut2004master1.epicgames.com/utmaster.openspy.net/" "${XDG_DATA_HOME}"/System/Default.ini
+    # TODO: Add foxWSFix support
+    #if grep -Fxq "InputClass=Class'foxWSFix.foxPlayerInput'" "${XDG_DATA_HOME}"/System/User.ini; then
+    #    echo "foxWSFix already installed"
+    #else
+    #    cp /app/extra/foxWSFix/System/foxWSFix.u "${XDG_DATA_HOME}"/System/foxWSFix.u
+    #    echo -e "\n\n[XGame.xPlayer]\nInputClass=Class'foxWSFix.foxPlayerInput'" >> "${XDG_DATA_HOME}"/System/User.ini
+    #fi
+    touch "${XDG_DATA_HOME}"/patched
+fi
+
+if [ ! -s "${XDG_DATA_HOME}/System/cdkey" ]; then
+    CDKEY=$(zenity --entry \
+            --width=400    \
+            --height=200   \
+            --title="CD Key Required" \
+            --text="Please enter your Unreal Tournament 2004 CD Key." \
+            --entry $CDKEY)
+    [ ! -z "$CDKEY" ] && echo $CDKEY > "${XDG_DATA_HOME}"/System/cdkey
+fi
+
+cd ${XDG_DATA_HOME}/System
+exec ./ut2004-bin-linux-amd64 "${@}"
```

Comment:
```suggestion
exec ./ut2004-bin-linux-amd64 "$@"
```

Reply:
I am not sure why `"${@}"` was also used in our ioquake3 Flatpak (since I am not the one who wrote the launcher script), but you are probably correct.